### ISSUE TITLE
Remove duplication in configurations between db-sync and node

### ISSUE THREAD

**Issue by kderme** [input-output-hk/cardano-db-sync#1117]
It seems there is a lot of duplication between db-sync and node code in the `NodeConfig` and `ProtocolInfo` department, which is also a big attraction to bugs. We should check if we can reduce it.

**Comment by erikd** [comment]
I have wanted to do this from the very beginning of `db-sync`.

**Comment by Cmdv** [comment]
is the primary idea here to align
https://github.com/input-output-hk/cardano-db-sync/blob/2b64878dab5165766730372e2ed3e47bae918360/cardano-db-sync/src/Cardano/DbSync/Config/Node.hs#L27-L49
with the types found in [this file](https://github.com/input-output-hk/cardano-node/blob/master/cardano-node/src/Cardano/Node/Types.hs) on the cardano-node repo?

**Comment by kderme** [comment]
Yes, I guess it's part of this file in Node. But some parts, like `TriggerHardFork` seem to miss from the node file.

**Comment by kderme** [comment]
I think the idea is to find the configuration types and parsers from Node and reuse them, instead of redifining them. This may not be trivial and the config may not match 1-1.

**Comment by erikd** [comment]
The problem is that, last time I looked (2 or more years ago) the configuration types and parsers in `node` did not really seem to be written to be reused. IMO (based on my last look) the `node` stuff would need to be improved and made reusable first.

**Comment by profd2004** [comment]
Closing this as part of a triage pass over the older open issues.

db-sync has read the node config directly by path since 6.0.0, so the duplication described here is gone.

Happy to reopen if I've missed something.
